pkg/config/app_config_test.go

```diff
@@ -50,9 +51,30 @@ git:
        - pattern: "^foo.bar*"
          replace: '[FUN $0] '`,
 		}, {
-			"Incomplete Configuration",
-			"git:",
-			"git:",
+			name:     "Incomplete Configuration",
+			input:    "git:",
+			expected: "git:",
+		}, {
+			name: "No changes made when already migrated",
+			input: `
+git:
+  commitPrefix:
+    - pattern: "Hello World"
+      replace: "Goodbye"
+  commitPrefixes:
+    foo:
+      - pattern: "^\\w+-\\w+.*"
+        replace: '[JIRA $0] '`,
+			expected: `
```

Comment:
This test is relying on the knowledge that our yaml rewriting will change the indentation to 4. If we change that as proposed in #4312, this test will no longer catch the error but succeed either way.

I think I would use some other indentation (e.g. 3 at top level, 1 at the second level) to make this more robust.